main.go

```diff
@@ -219,6 +220,107 @@ func importFile(projectId string, file, mimetype string, ec runner.EvalContext)
 	}, nil
 }
 
+func runQuery(queryRaw string, project *runner.ProjectState, ec *runner.EvalContext, args *args, files []string) error {
+	query := rewriteQuery(queryRaw)
+	panel := &runner.PanelInfo{
+		Type:    runner.DatabasePanel,
+		Content: query,
+		Id:      uuid.New().String(),
+		Name:    uuid.New().String(),
+		DatabasePanelInfo: &runner.DatabasePanelInfo{
+			Database: runner.DatabasePanelInfoDatabase{
+				ConnectorId: project.Connectors[0].Id,
+			},
+		},
+	}
+
+	err := ec.EvalDatabasePanel(project, 0, panel, nil, args.cacheSettings)
+	if err != nil {
+		if e, ok := err.(*runner.DSError); ok && e.Name == "NotAnArrayOfObjectsError" {
+			rest := "."
+			nth, err := strconv.Atoi(e.TargetPanelId)
+			if err == nil {
+				rest = ": " + files[nth] + "."
+			}
+			return fmt.Errorf("Input is not an array of objects%s\n", rest)
+		}
+	}
+
+	resultFile := ec.GetPanelResultsFile(project.Id, panel.Id)
+	return dumpJSONFile(resultFile, args.pretty, args.schema)
+}
+
+func repl(project *runner.ProjectState, ec *runner.EvalContext, args *args, files []string) error {
+
+	tempfile, err := ioutil.TempFile("", "dsq-hist")
+	if err != nil {
+		return err
+	}
+
+	completer := readline.NewPrefixCompleter(
+		readline.PcItem("SELECT"),
+		readline.PcItem("FROM"),
+		readline.PcItem("WHERE"),
+		readline.PcItem("AND"),
+		readline.PcItem("OR"),
+		readline.PcItem("IN"),
+		readline.PcItem("JOIN"),
+	)
+
+	filterInput := func(r rune) (rune, bool) {
+		switch r {
+		// block CtrlZ feature
+		case readline.CharCtrlZ:
+			return r, false
+		}
+		return r, true
+	}
+
+	defer os.Remove(tempfile.Name())
```

Comment:
Just a nit, I normally prefer to keep cleanup code right after initialize code. I'd put this at line 259